### PR TITLE
 Problem: (CRO-355) No limit on number of requests in a batch RPC call to tendermint

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Divided batch RPC calls in chunks of size 100 requests per call

### CODE REVIEW

**Comment by devashishdxt** [comment]
@tomtau: Is a limit of 100 on the number of requests per batch call ok or should we reduce it further (just to be safe)?

File: client-index/src/synchronizer.rs [hunk 1]
```diff
@@ -46,34 +48,37 @@ where
...
 
-        let blocks = self
-            .client
-            .block_batch((last_block_height + 1)..=current_block_height)?;
+        // Send batch RPC requests to tendermint in chunks of 100 requests per batch call
+        for chunk in ((last_block_height + 1)..=current_block_height)
+            .chunks(100)
+            .into_iter()
+        {
+            let range = chunk.collect::<Vec<u64>>();
```
[the remark this answers is not in the document]
Comment (devashishdxt) [reply client-index/src/synchronizer.rs]: We only have one `chunk` object and we need to pass it into multiple functions. `Chunk` does not implement `Clone`, so I created a `Vec` and passed iterators over the vector in both functions.

**Comment by tomtau** [comment]
> @tomtau: Is a limit of 100 on the number of requests per batch call ok or should we reduce it further (just to be safe)?

I think it's fine -- RPC shouldn't be exposed directly to the internet, so ideally, there'll be some rate-limiting proxy that would ensure some sort of QoS on public nodes.
It seems Tendermint doesn't have a hard limit on this -- for other query-like things, the limit I've seen were that 50 was the default, 100 was the maximum; on the "blockchain" command (returns block headers from minheight to maxheight), it's 20 most recent items.
We can do some benchmarking or ask on the Slack channel if 100 batches are safe.

File: client-index/src/synchronizer.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomtau) [reply client-index/src/synchronizer.rs]: `&chunk.iter()` / `&chunk.into_iter()` ?

[the remark this answers is not in the document]
Comment (devashishdxt) [reply client-index/src/synchronizer.rs]: I tried. But, non-mutable reference to an iterator does not implement `Iterator` trait.

**Pull request merged**